### PR TITLE
Add support for GitLab Push Event

### PR DESCRIPTION
**Pull request by vanloswang**
Enable WebHook for build intergrated with GitLab Push Event, fix https://github.com/openshift/origin/issues/11002

### CODE REVIEW

**Comment by mfojtik** [comment]
LGTM @bparees ?

File: pkg/build/webhook/github/github.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 )
 
 var (
-	ErrNoGitHubEvent = errors.New("missing X-GitHub-Event or X-Gogs-Event")
+	ErrNoGitHubEvent = errors.New("missing X-GitHub-Event or X-Gogs-Event or X-Gitlab-Event")
```
Comment (bparees) [inline pkg/build/webhook/github/github.go]: please change this text to:
"missing X-GitHub-Event, X-Gogs-Event, or X-Gitlab-Event"

File: pkg/build/webhook/github/github.go [hunk 2]
```diff
@@ -57,8 +57,8 @@ func (p *WebHook) Extract(buildCfg *api.BuildConfig, secret, path string, req *h
 		return revision, envvars, proceed, err
 	}
 	method := getEvent(req.Header)
-	if method != "ping" && method != "push" {
-		return revision, envvars, proceed, fmt.Errorf("Unknown X-GitHub-Event or X-Gogs-Event %s", method)
+	if method != "ping" && method != "push" && method != "Push Hook" {
+		return revision, envvars, proceed, fmt.Errorf("Unknown X-GitHub-Event or X-Gogs-Event or X-Gitlab-Event %s", method)
```
Comment (bparees) [inline pkg/build/webhook/github/github.go]: same message change here (comma separated list instead of multiple ors)

**Comment by vanloswang** [comment]
@bparees Changed the notice msg.

**Comment by bparees** [comment]
please squash your commits and lgtm.

**Comment by bparees** [comment]
@vanloswang can you confirm that the gitlab webhok payload is compatible w/ the github push payload?

**Comment by bparees** [comment]
@vanloswang please squash and i'll merge based on your confirmation in the related issue that this fix works.

**Comment by vanloswang** [comment]
@bparees goto https://github.com/openshift/origin/pull/11031

**Pull request closed without merging**